fdbserver/MetricLogger.actor.cpp

```diff
@@ -364,41 +375,275 @@ ACTOR Future<Void> updateMetricRegistration(Database cx, MetricsConfig* config,
-	// Never log to an empty prefix, it's pretty much always a bad idea.
-	if (prefix.size() == 0) {
-		TraceEvent(SevWarnAlways, "TDMetricsRefusingEmptyPrefix").log();
-		return Void();
+// ACTOR Future<Void> runMetrics(Future<Database> fcx, Key prefix) {
+// 	// Never log to an empty prefix, it's pretty much always a bad idea.
+// 	if (prefix.size() == 0) {
+// 		TraceEvent(SevWarnAlways, "TDMetricsRefusingEmptyPrefix").log();
+// 		return Void();
+// 	}
+
+// 	// Wait until the collection has been created and initialized.
+// 	state TDMetricCollection* metrics = nullptr;
+// 	loop {
+// 		metrics = TDMetricCollection::getTDMetrics();
+// 		if (metrics != nullptr)
+// 			if (metrics->init())
+// 				break;
+// 		wait(delay(1.0));
+// 	}
+
+// 	state MetricsConfig config(prefix);
+
+// 	try {
+// 		Database cx = wait(fcx);
+// 		Future<Void> conf = metricRuleUpdater(cx, &config, metrics);
+// 		Future<Void> dump = dumpMetrics(cx, &config, metrics);
+// 		Future<Void> reg = updateMetricRegistration(cx, &config, metrics);
+
+// 		wait(conf || dump || reg);
+// 	} catch (Error& e) {
+// 		if (e.code() != error_code_actor_cancelled) {
+// 			// Disable all metrics
+// 			for (auto& it : metrics->metricMap)
+// 				it.value->setConfig(false);
+// 		}
+
+// 		TraceEvent(SevWarnAlways, "TDMetricsStopped").error(e);
+// 		throw e;
+// 	}
+// 	return Void();
+// }
+
+ACTOR Future<Void> startMetricsSimulationServer(MetricsDataModel model) {
+	if (model == MetricsDataModel::NONE) {
+		return Void{};
 	}
+	state uint32_t port = (model == STATSD) ? FLOW_KNOBS->STATSD_UDP_EMISSION_PORT : FLOW_KNOBS->OTEL_UDP_EMISSION_PORT;
+	TraceEvent(SevInfo, "MetricsUDPServerStarted").detail("Address", "127.0.0.1").detail("Port", port);
+	state NetworkAddress localAddress = NetworkAddress::parse("127.0.0.1:" + std::to_string(port));
+	state Reference<IUDPSocket> serverSocket = wait(INetworkConnections::net()->createUDPSocket(localAddress));
+	serverSocket->bind(localAddress);
+	state Standalone<StringRef> packetString = makeString(IUDPSocket::MAX_PACKET_SIZE);
+	state uint8_t* packet = mutateString(packetString);
 
-	// Wait until the collection has been created and initialized.
-	state TDMetricCollection* metrics = nullptr;
 	loop {
-		metrics = TDMetricCollection::getTDMetrics();
-		if (metrics != nullptr)
-			if (metrics->init())
-				break;
-		wait(delay(1.0));
+		int size = wait(serverSocket->receive(packet, packet + IUDPSocket::MAX_PACKET_SIZE));
+		auto message = packetString.substr(0, size);
+
+		// Let's just focus on statsd for now. For statsd, the message is expected to be seperated by newlines. We need
+		// to break each statsd metric and verify them individually.
+		if (model == MetricsDataModel::STATSD) {
+			std::string statsd_message = message.toString();
+			auto metrics = splitString(statsd_message, "\n");
+			for (const auto& metric : metrics) {
+				ASSERT(verifyStatsdMessage(metric));
+			}
+		}
```

Comment:
Can you add an OTEL test here as well, now that you support OTEL as well. It can be simple, but we should have something.